relay.go

```diff
@@ -579,16 +584,16 @@ func (r *Relayer) handleNonCallReq(f *Frame) error {
 	originalID := f.Header.ID
 	f.Header.ID = item.remapID
 
-	sent, failure := item.destination.Receive(f, frameType)
+	sent, failure, dropped := item.destination.Receive(f, frameType)
```

Comment:
Do we need a third return? A couple of alternative thoughts:
1. Isn't `sent` a sign of whether the frame is handled by the receiver, and so indicates we didn't drop it. is `!sent` == dropped already?
2. Another option is to treat `Receive` as transfer of ownership, regardless of success or failure. So the frame belongs to `item.destination`. In the case where the channel has space, it ends up going to `writeFrames` which then puts it back in the frame pool. However, in the error case, we could call `c.opts.FramePool.Release(frame)`

(2) is simple, but adds one extra caller to release frames outside of read/write frames (it's a little cleaner + easier to follow to have limited interactions with pooling), but it might still be better than changing the Receive API.

Maybe (1) is the best option?